cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,248 @@
+
+Pytest test configuration file.
+
+"""
+import pytest
+from pathlib import Path
+from subprocess import run, TimeoutExpired
+from time import sleep
+import json
+import os
+import shutil
+
+@pytest.fixture(scope="session")
+def installer_path(request):
+    return Path(request.config.getoption("--installer-path")).resolve()
+
+@pytest.fixture(scope="session")
+def install_root(request):
+    return Path(request.config.getoption("--install-root")).resolve()
+
+@pytest.fixture(scope="session")
+def project_path(request):
+    # MSBuild doesn't want us to output in the temp directory MSB8029
+    # so we use a pre-defined folder outside of %TMP%
+    #return tmpdir_factory.mktemp("TestProject")
+    path = Path(request.config.getoption("--project-path")).resolve()
+    yield path
+
+    # wait a few seconds for processes to quit
+    sleep(5)
+
+    shutil.rmtree(path)
+
+@pytest.fixture(scope="session")
+def cmake_path(install_root):
+    return install_root / 'cmake' / 'runtime' / 'cmake-3.20.2-windows-x86_64' / 'bin' / 'cmake.exe'
+
+@pytest.fixture(scope="session")
+def test_installer(installer_path, install_root):
+    assert installer_path.is_file(), f"Invalid installer path {installer_path}"
+
+    # when the installer is run with timeout of 30 minutes
+    result = run([installer_path , f"InstallFolder={install_root}", "/quiet"], timeout=30*60)
+
+    # the installer succeeds
+    assert result.returncode == 0, f"Installer failed with exit code {result.returncode}"
+
+    # the install root is created
+    assert install_root.is_dir(), f"Invalid install root {install_root}"
+
+@pytest.mark.parametrize(
+    "filename", [
+        pytest.param('Editor.exe'),
+        pytest.param('AssetProcessor.exe'),
+        pytest.param('MaterialEditor.exe'),
+        pytest.param('o3de.exe'),
+    ]
+)
+def test_binaries_exist(test_installer, install_root, filename):
+    bin_folder = install_root / 'bin/Windows/profile/Default'
+    assert (bin_folder / filename).is_file(), f"{filename} not found in {bin_folder}"
+
+
+@pytest.fixture(scope="session")
+def test_o3de_registers_engine_fixture(install_root, test_installer):
+    try:
+        bin_folder = install_root / 'bin' / 'Windows' / 'profile' / 'Default'
+        result = run([str(bin_folder / 'o3de.exe')], timeout=7)
+
+        # o3de.exe should not close with an error code 
+        assert result.returncode == 0, f"o3de.exe failed with exit code {result.returncode}"
+    except TimeoutExpired as e:
+        # we expect to close the app ourselves
+        pass
+
+    # the engine is registered
+    engine_json_path = Path(install_root).resolve() / 'engine.json'
+    with engine_json_path.open('r') as f:
+        engine_json_data = json.load(f)
+
+    engine_name = engine_json_data['engine_name']
+
+    manifest_path = Path(os.path.expanduser("~")).resolve() / '.o3de' / 'o3de_manifest.json'
+    with manifest_path.open('r') as f:
+        manifest_json_data = json.load(f)
```

Comment:
Is there not an option to retrieve the engine name from the o3de CLI?

Reply:
I believe there is, but it may only check the 'engines' section in the manifest and not the 'engines_path' - will confirm